### ISSUE TITLE
Update all the diagrams with SVG versions

### ISSUE THREAD

**Issue by stevepiercy** [Pylons/pyramid#1492]
All diagrams should complement each other and be consistent in design. Some may be "zoom" or "detail" views, kind of like an inset on paper maps.

Here's the repo where I push each version of a diagram until it is ready for a PR to the Pyramid repo:
https://github.com/stevepiercy/pyramid_request_diagram

Here are all the diagrams that I have in my queue:
- [x] [Request Processing](https://github.com/stevepiercy/pyramid_request_diagram/blob/master/pyramid_request_processing.svg) (See #1472)
- [ ] [Request Processing with Exceptions](https://github.com/stevepiercy/pyramid_request_diagram/blob/master/pyramid_request_processing_with_exceptions.svg)
- [x] [Router](https://github.com/stevepiercy/pyramid_request_diagram/blob/master/pyramid_router.svg) (See #1479)
- [ ] [Traversal](https://github.com/stevepiercy/pyramid_request_diagram/blob/master/pyramid_traversal.svg)
- [ ] [Pyramid Onion](https://github.com/stevepiercy/pyramid_request_diagram/blob/master/pyramid_onion.svg)
- [ ] [Pyramid excview_tween](https://github.com/stevepiercy/pyramid_request_diagram/blob/master/pyramid_excview_tween.svg)

**Comment by jsivak** [comment]
Excellent work on the diagrams (they help immensely).

An issue:  The diagrams appear to look correct in Chrome and Opera, but Firefox is showing text running outside of some boxes and also overlapping text in some boxes.

Is this a known issue/concern?

**Comment by stevepiercy** [comment]
@jsivak need more info. They all look fine to me in latest version of FF.

**Comment by bertjwregeer** [comment]
https://github.com/stevepiercy/pyramid_request_diagram/blob/master/pyramid_onion.svg

Router executes the tweens...

WSGI -> Router -> Tweens -> View -> Tweens -> Router -> WSGI

**Comment by stevepiercy** [comment]
@bertjwregeer I currently have three tweens, `toolbar_tween`, `excview_tween`, and `tm_tween`. Should I keep them separate or combine them into the single "Tweens"?

**Comment by bertjwregeer** [comment]
toolbar_tween is only ever there when the toolbar is active. It is registered by the user. excview_tween you can leave, it is always there. tm_tween will soon wrap all of the above, but it too is only loaded when it is registered by the user.

**Comment by stevepiercy** [comment]
Thanks, that helps a lot. I think I can come up with a visualization of optional tweens.

**Comment by stevepiercy** [comment]
@bertjwregeer well, I toyed around with visual concepts, but none worked, so I just went with "(optional)" https://github.com/stevepiercy/pyramid_request_diagram/blob/master/pyramid_onion.svg

**Comment by jsivak** [comment]
@stevepiercy I'm checking from home now and you're right (no display problems), they look correct. I saw the issue from work; will have to re-check it on Monday.

**Comment by jsivak** [comment]
@stevepiercy , you're right, the SVG images are fine.. I had a corrupt profile in my Firefox at work.

**Comment by miketheman** [comment]
Huge fan of diagrams (I'm a visual learner).
I'm digging into Request Processing with Pyramid 2.0, and am trying to understand where the newer 2.0 Security Policy is invoked (vs the deprecated authentication and authorization policies), and can't exactly determine if it happens at the same spot (first in view pipeline) or not. Does the diagram need an update, or does [the document](https://docs.pylonsproject.org/projects/pyramid/en/latest/narr/router.html) need to update the language?

**Comment by luhn** [comment]
Yeah, both the legacy authz/authn policy and the new security policy use the same `secured_view` view deriver, so it's in the same spot in the pipeline.
